lib/longleaf/candidates/single_digest_provider.rb

```diff
@@ -0,0 +1,13 @@
+module Longleaf
+  # Provides a single set of digests for files
+  class SingleDigestProvider
```

Comment:
The `-f` option can supply multiple paths, or directories of paths, so the SingleDigestProvider would provide the same set of checksums for all the paths selected. Its a pretty unlikely case, I'm just not sure how to reconcile the that `--checkums` only takes one set of checksums with the behavior of `-f`. I guess I could do some validation of `-f` to check to see if there are multiple values or the value is a directory rather than having a separate digest provider. Do you think that would make more sense?

I had considered there could be digest providers in the future that did stuff like calculate the digest during registration

Reply:
I don't feel really strongly about this, but I did notice that the SingleDigestProvider accepts a file path presumably for consistency's sake and doesn't really do anything very unique. I started searching around about ruby interfaces before leaving this comment (more from my C++ background) :-) I think it's fine if you think this is the best approach.